src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -1,115 +1,123 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+
+
+
+/** Handles fetching and saving {@link Message} instances. */
+@WebServlet("/messages")
+public class MessageServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+
+  private float getSentimentScore(String text) throws IOException {
+    Document doc = Document.newBuilder()
+            .setContent(text).setType(Type.PLAIN_TEXT).build();
+
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    languageService.close();
+
+    return sentiment.getScore();
+  }
+
+
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
+   * an empty array if the user is not provided.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+
+    String user = request.getParameter("user");
+
+    if (user == null || user.equals("")) {
+      // Request is invalid, return empty array
+      response.getWriter().println("[]");
+      return;
+    }
+
+    List<Message> messages = datastore.getMessages(user);
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+
+    response.getWriter().println(json);
+  }
+
+  /** Stores a new {@link Message}. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    String user = userService.getCurrentUser().getEmail();
+
+    // basicWithImages allows a, b, blockquote, br, cite, code, dd, dl, dt, em, i, li, ol,
+    // p, pre, q, small, span, strike, strong, sub, sup, u, ul, and image tags
+    //String userText = Jsoup.clean(request.getParameter("text"), Whitelist.basicWithImages());
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String recipient = request.getParameter("recipient");   
+      
+ 
+    String regex = "(https?://([^\\s.]+.?[^\\s.])+/([^\\s.]+.?[^\\s.])+.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+    System.out.println(textWithImagesReplaced);
+    
+        
+    Message message = new Message(user, textWithImagesReplaced, 0.0f , recipient);

```

Comment:
We want to pass the sentiment here instead of 0.0f can you bring that code back.